### PR TITLE
Path conditions

### PR DESCRIPTION
**Pull request by CPASimUSante**
Please review this modifications.
With these modifications 
at the path level you can define wether the conditions applies on the next step only or on every step after : 
For a path with  4 steps, and a condition on step 2 : in the first case, the user missing the condition can't go on step 3, in the second case, the user can't access step 3 and 4 (and further if any)

Needs testing with real paths.

### CODE REVIEW

**Comment by stefk** [comment]
@CPASimUSante, the weird build errors are probably related to recent changes on the master branch which are not present in your branch. Can you update your PR (there should be an "Update branch" button below)?

**Comment by CPASimUSante** [comment]
ok, i clicked on "update branch" button
Wait and see

File: plugin/path/Entity/Path/Path.php [hunk 1]
```diff
@@ -80,12 +80,22 @@ class Path extends AbstractResource implements PathInterface, \JsonSerializable
     protected $description;
 
     /**
+     * Does the path manage conditions that block all the further steps or just the next one
+     *
+     * @var boolean
+     *
+     * @ORM\Column(name="is_complete_blocking_condition", type="boolean")
+     */
+    protected $isCompleteBlockingCondition;
```
Comment (Elorfin) [inline plugin/path/Entity/Path/Path.php]: Boolean properties must not start with 'is' or 'has', use `completeBlockingCondition` (you don't need to rename the DB column so you don't have to generate migrations again).

File: plugin/path/Entity/Path/Path.php [hunk 2]
```diff
@@ -282,6 +292,30 @@ public function hasBreadcrumbs()
     }
 
     /**
+     * Get isCompleteBlockingCondition
+     *
+     * @return boolean
+     */
+    public function getIsCompleteBlockingCondition()
```
Comment (Elorfin) [inline plugin/path/Entity/Path/Path.php]: just name your method `isCompleteBlockingCondition()`, don't need the get (Symfony also finds getters if they start by is or has).

File: plugin/path/Entity/Path/Path.php [hunk 3]
```diff
@@ -282,6 +292,30 @@ public function hasBreadcrumbs()
...
+    }
+
+    /**
+     * Set isCompleteBlockingCondition
+     *
+     * @param boolean $isCompleteBlockingCondition
+     *
+     * @return \Innova\PathBundle\Entity\Path\Path
+     */
+    public function setIsCompleteBlockingCondition($isCompleteBlockingCondition)
```
Comment (Elorfin) [inline plugin/path/Entity/Path/Path.php]: Becomes `setCompleteBlockingCondition`

File: plugin/path/Resources/public/js/Path/Controller/PathBaseCtrl.js [hunk 4]
```diff
@@ -11,10 +11,12 @@ var PathBaseCtrl = function PathBaseCtrl($window, $route, $routeParams, PathServ
     // Store path to make it available by all UI components
     this.pathService.setId(this.id);
     this.pathService.setPath(this.path);
+    this.pathService.setIsCompleteBlockingCondition(this.path.isCompleteBlockingCondition);
```
Comment (Elorfin) [inline plugin/path/Resources/public/js/Path/Controller/PathBaseCtrl.js]: same here, remove the `is` from method.

File: plugin/path/Resources/public/js/Path/Controller/PathBaseCtrl.js [hunk 5]
```diff
@@ -11,10 +11,12 @@ var PathBaseCtrl = function PathBaseCtrl($window, $route, $routeParams, PathServ
...
 
     this.currentStep = $routeParams;
 
     this.summaryState = this.pathService.getSummaryState();
+    this.isCompleteBlockingCondition = this.pathService.getIsCompleteBlockingCondition();
```
Comment (Elorfin) [inline plugin/path/Resources/public/js/Path/Controller/PathBaseCtrl.js]: `this.pathService.isCompleteBlockingCondition()`

File: plugin/path/Resources/public/js/Path/Service/PathService.js [hunk 6]
```diff
@@ -39,6 +39,12 @@
             };
 
             /**
+             * is condition blocking all children of a step
+             * @type {boolean}
+             */
+            var isCompleteBlockingCondition = true;
```
Comment (Elorfin) [inline plugin/path/Resources/public/js/Path/Service/PathService.js]: same remark than for PHP : no `is` in property name.

**Comment by Elorfin** [comment]
I've added some little comments about coding standards.
I'm sorry, I will have no time today to check if the code really work as expected : I will give it a try next week.

**Comment by stefk** [comment]
@CPASimUSante, can you update again?

**Comment by Elorfin** [comment]
LGTM

**Comment by stefk** [comment]
LGTM

**Pull request merged**